ons_alpha/taxonomy/models.py

```diff
@@ -77,6 +77,12 @@ def clean(self):
         if not parent and instance.get_parent() != parent:
             raise ValidationError("Cannot make a child topic a top-level topic.")
 
+        # depth starts with one. We cannot rely on instance.get_parent()
+        # until after the topic is created, and parent can be None
+        depth = parent.depth + 1 if parent else 1
+        if Topic.objects.filter(name=cleaned_data["name"], depth=depth).exists():
```

Comment:
Would `if self.instance.get_siblings().exclude(id=self.instance.id).filter(name=cleaned_data["name"]).exists()` be better? This avoids the need to calculate the depth.